### PR TITLE
[Chore](executor) do not retry query when catched UserException and cloud mode is off

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
do not retry query when catched UserException and cloud mode is off

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -848,7 +848,7 @@ private void handleQueryWithRetry(TUniqueId queryId) throws Exception {
                 LOG.debug("due to exception {} retry {} rpc {} user {}",
                         e.getMessage(), i, e instanceof RpcException, e instanceof UserException);
                 String msg = e.getMessage();
-                boolean isNeedRetry = true;
+                boolean isNeedRetry;
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:851]: Make it default false?

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

File: be/src/pipeline/local_exchange/local_exchanger.cpp [hunk 2]
```diff
@@ -302,24 +302,15 @@ Status LocalMergeSortExchanger::build_merger(RuntimeState* state,
...
-                    if (_free_block_limit == 0 ||
-                        _free_blocks.size_approx() < _free_block_limit * _num_sources) {
-                        _free_blocks.enqueue(std::move(next_block));
-                    }
-                    sub_mem_usage(local_state, id, block->allocated_bytes());
-                } else {
-                    *eos = true;
-                }
-            } else if (_data_queue[id].try_dequeue(next_block)) {
+            if (_data_queue[id].try_dequeue(next_block)) {
```
Comment (Gabriel39) [inline be/src/pipeline/local_exchange/local_exchanger.cpp:305]: Consider events below:
1. `_data_queue[id].try_dequeue(next_block)` return false which means no data in this queue
2. Another block is push into `_data_queue[id]` and sub `_running_sink_operators` to 0.
3. `_running_sink_operators == 0` is true and set eos to true.
And then the result is wrong

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**